tex/gregoriotex.lua

```diff
@@ -1237,119 +1203,107 @@ local function locate_file(filename)
   return result
 end
 
-local function include_score(input_file, force_gabccompile, allow_deprecated)
-  if string.match(input_file, "[#%%]") then
+local function include_score(gabc_file, force_gabccompile, allow_deprecated)
+  local sep = ""
+  local onwindows = (os.type == "windows" or
+                     string.find(os.getenv("PATH"),";",1,true))
+  if onwindows then
+    sep = "\\"
+  else
+    sep = "/"
+  end
+    
+  if string.match(gabc_file, "[#%%]") then
     err("GABC filename contains invalid character(s): # %%\n"
-        .."Rename the file and retry: %s", input_file)
+        .."Rename the file and retry: %s", gabc_file)
   end
-  local has_extention = false
-  local file_dir,input_name
+  local gabc_dir, base
   local extensions = {['gabc']=true, ['gtex']=true, ['tex']=true}
-  if extensions[string.match(input_file, "([^%.\\/]*)$")] then
-    has_extention = true
-  end
-  if has_extention then
-    file_dir,input_name = string.match(input_file, "(.-)([^\\/]-)%.?[^%.\\/]*$")
+  if extensions[string.match(gabc_file, "([^%.\\/]*)$")] then
+    gabc_dir, base = string.match(gabc_file, "(.-)([^\\/]-)%.?[^%.\\/]*$")
   else
-    file_dir,input_name = string.match(input_file, "(.-)([^\\/]*)$")
-  end
-
-  local cleaned_filename = input_name:gsub("[%s%+%&%*%?$@:;!\"\'`]", "-")
-  local gabc_filename = string.format("%s%s.gabc", file_dir, input_name)
-  local gabc_file = locate_file(gabc_filename)
-  local gtex_filename = string.format("%s%s-%s.gtex", file_dir, cleaned_filename,
-      internalversion:gsub("%.", "_"))
-  local gtex_file = locate_file(gtex_filename)
-  local glog_file = string.format("%s%s-%s.glog", file_dir, cleaned_filename,
-      internalversion:gsub("%.", "_"))
-  if not gtex_file then
-    clean_old_gtex_files(file_dir..cleaned_filename)
-    log("The file %s does not exist. Will use gabc file", gtex_filename)
-    if gabc_file then
-      local gabc = io.open(gabc_file, 'r')
-      if gabc == nil then
-        err("\n Unable to open %s", gabc_file)
-        return
-      else
-        gabc:close()
-      end
-      local sep = ""
-      local onwindows = os.type == "windows" or
-        string.find(os.getenv("PATH"),";",1,true)
-      if onwindows then
-        sep = "\\"
-      else
-        sep = "/"
-      end
-      local output_dir = base_output_dir..sep..file_dir
-      info(output_dir)
-      if not lfs.exists(output_dir) then
-        if not lfs.exists(base_output_dir) then
-          lfs.mkdir(base_output_dir)
-        end
-        local err,message = lfs.rmkdir(output_dir)
-        if not err then
-          info(message)
-        end
+    gabc_dir, base = string.match(gabc_file, "(.-)([^\\/]*)$")
+  end
+  local base_cleaned = base:gsub("[%s%+%&%*%?$@:;!\"\'`]", "-")
+
+  -- Find gabc file
+  gabc_file = string.format("%s%s.gabc", gabc_dir, base)
+  local gabc_found = locate_file(gabc_file)
+  
+  -- Set up output directory
+  local output_dir = base_output_dir..sep..gabc_dir
```

Comment:
`sep` appears to be blank on Windows here

Reply:
That's very mysterious, but as you suggested, maybe the problem can be circumvented.